native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/states/http2/SenderState.java

```diff
@@ -54,6 +54,13 @@
     void writeOutboundRequestBody(ChannelHandlerContext ctx, HttpContent httpContent,
                                   Http2MessageStateContext http2MessageStateContext) throws Http2Exception;
 
+    /**
+     * Writes a reset frame to reset the stream.
+     *
+     * @param ctx                      the channel handler context
+     */
+    void writeRstStream(ChannelHandlerContext ctx);
```

Comment:
Let's reuse the `writeOutboundRequestBody()` of the `SendingRstFrame` class rather introducing another action. Otherwise this action is spanned across all the other states unnecessarily.

Reply:
Added the changes in https://github.com/ballerina-platform/module-ballerina-http/pull/659/commits/287830336fb2b11641e3547b9ee16b050e076d95